compiler/rustc_error_codes/src/error_codes/E0755.md

```diff
@@ -0,0 +1,27 @@
+The `ffi_pure` attribute was used on a non-foreign function.
+
+Erroneous code example:
+
+```compile_fail,E0755
+#![feature(ffi_pure)]
+
+#[ffi_pure] // error!
+pub fn foo() {}
+# fn main() {}
+```
+
+The `ffi_pure` attribute can only be used on foreign functions which do not have
+effects except for their return value. You can find more information about it in
+the [unstable Rust Book].
+
+```
+#![feature(ffi_pure)]
+
+extern "C" {
+    #[ffi_pure] // ok!
+    pub fn strlen(s: *const i8) -> isize;
+}
```

Comment:
I believe this also work right?
```rust
#[ffi_pure]
extern "C" pub fn foo() {}
```

Reply:
I think it's better to have a bigger contrast. :)

Reply:
Oh, I was just suggesting to show both methods so the readers know.